### PR TITLE
Handle -ve rotations, null overlap in sm/DC

### PR DESCRIPTION
**Pull request by ndevenish**
**JIRA ticket**: [LIMS-1232](https://jira.diamond.ac.uk/browse/LIMS-1232)

**Summary**:

I19 is now doing small molecule serial, where every other collection is run with backwards oscillation. Unfortunately, recording this accurately in ISPyB doesn't show the autoprocessing sections in Synchweb.

This changes the check to be truthy (nonzero) instead of purely positive.

Also fixes an adjacent issue where a null value for overlap was passing the intended non-zero check (null != 0). This is made truthy also.

**Changes**:
- Altered data checks to check truthiness instead of comparisons to zero

**To test**:
- Viewing https://ispyb.diamond.ac.uk/dc/visit/cy35300-1/dcg/11242587 should show "Auto Processing" for both collections, instead of the current situation:
image

### CODE REVIEW

File: client/src/js/templates/types/sm/dc/dc.html [hunk 1]
```diff
@@ -43,8 +43,8 @@ <h1 class="title"></h1>
     </ul>
     
     <div class="holder">
-    <% if (AXISRANGE > 0) { %>
-        <% if (OVERLAP != 0) { %>
+    <% if (AXISRANGE) { %>
```
Comment (ndg63276) [inline client/src/js/templates/types/sm/dc/dc.html:46]:
```suggestion
    <% if (parseFloat(AXISRANGE)) { %>
```

File: client/src/js/templates/types/sm/dc/dc.html [hunk 2]
```diff
@@ -43,8 +43,8 @@ <h1 class="title"></h1>
...
+        <% if (OVERLAP) { %>
```
Comment (ndg63276) [inline client/src/js/templates/types/sm/dc/dc.html:47]:
```suggestion
        <% if (parseFloat(OVERLAP)) { %>
```

**Review by ndg63276 — CHANGES_REQUESTED** [review]
The values appear to be strings, so even zeroes are truthy.

**Comment by ndevenish** [comment]
This appears to work fine!
image

**Review by ndg63276 — APPROVED** [review]

**Pull request merged**